### ISSUE TITLE
Allow selection of dbt profile target from command line or ENV var

### ISSUE THREAD

**Issue by gnilrets** [sqlfluff/sqlfluff#1708]
I use SQLFluff during dbt development when running against a `dev` target dbt profile.  I also use SQLFluff during a CI process that runs dbt against a `test` target dbt profile.  SQLFluff doesn't allow me to choose which profile I want to use in these different environments.  I would really like to be able to control which profile SQLFluff used using a command line argument or an environment variable.

**Comment by BeadW** [comment]
Good Idea. Currently we work around this by having our CI use `sed` to replace the target in profiles.yml but it is not a great solution.

**Comment by johalnes** [comment]
Is there any news regarding this issue? 🙂

**Comment by vglocus** [comment]
I would too like something similar to this.
I am using this with the dbt templater.

E.g. support some `get_env(var_name, default)` ability in sqlfluff config.

Two current scenarios:
* Select profile/target as we have macros and other logic where the output is determined by these settings
* Alter dbt variable values without altering the sqlfluff config file. We have a common partition variable used to determine what partition(s) are being processed,  and it is not uncommon to have logic based on `{% if var('running_partition') > "2024-04-01" %}`. It would be great to be able to alter var values (given in the context setting) without altering the config file.

**Comment by ota2000** [comment]
Hi, I'd like to work on this.

We have a use case where we need different `[sqlfluff:templater:dbt]` `target` values for local development (`local`) and CI (`dev`). Currently we maintain two separate config files (`.sqlfluff` and `.sqlfluff-ci`) that are identical except for the `target` value, and switch between them with `--config`.

I'm thinking of implementing a general-purpose `--config-override` CLI flag that allows overriding any config value:

```bash
sqlfluff lint --config-override templater:dbt:target=dev
```

This would fit naturally into the existing `overrides` mechanism in `FluffConfig.__init__()` by extending it to support sections beyond `core`. It would also solve the original request in this issue without being too narrow (e.g., a dbt-specific `--dbt-target` flag).

Would this approach be acceptable? Happy to open a PR if so.
